src/mscorlib/shared/System/Diagnostics/Tracing/TraceLogging/DataCollector.cs

```diff
@@ -143,6 +143,51 @@ internal void AddBinary(string value, int size)
             }
         }
 
+        internal unsafe void AddNullTerminatedString(string value)
+        {
+            // Don't use value.Length here because string allows for embedded NULL characters.
+            int size = 0;
+            fixed (char* strValue = value)
+            {
+                char* pChar = strValue;
+                do
+                {
+                    // Char == UTF16
+                    size += 2;
+                }
+                while (*pChar++ != '\0');
+            }
+
+            if (size > ushort.MaxValue)
+            {
+                size = ushort.MaxValue - 1;
+            }
+
+            if (this.bufferNesting != 0)
+            {
+                this.EnsureBuffer(size);
+            }
+
+            if (size != 0)
+            {
+                if (this.bufferNesting == 0)
+                {
+                    this.ScalarsEnd();
+                    this.PinArray(value, size);
+                }
+                else
+                {
+                    var oldPos = this.bufferPos;
+                    this.bufferPos = checked(this.bufferPos + size);
```

Comment:
Make sure we have some tests that have strings in nested structures (we probably do) this insures that we tests both of these code paths.

Reply:
I have confirmed that we do have tests for this and that they are passing.